### PR TITLE
sockopt add OpenBSD's SO_RTABLE to set the route table.

### PR DESCRIPTION
**Pull request by devnexen**
Also SO_ACCEPTFILTER for FreeBSD/NetBSD.

### CODE REVIEW

File: src/sys/socket/sockopt.rs [hunk 1]
```diff
@@ -766,6 +766,26 @@ sockopt_impl!(
...
+    libc::SOL_SOCKET,
+    libc::SO_RTABLE,
+    i32
+);
+#[cfg(any(target_os = "freebsd", target_os = "netbsd"))]
+sockopt_impl!(
+    /// Get/set a filter on this socket before accepting connections similarly
+    /// to Linux's TCP_DEFER_ACCEPT but after the listen's call.
+    AcceptFilter,
+    Both,
```
Comment (asomers) [inline src/sys/socket/sockopt.rs:784]: This is set-only, at least on FreeBSD.

**Review by asomers — CHANGES_REQUESTED** [review]

File: src/sys/socket/sockopt.rs (hunk 1 above)
Comment (devnexen) [reply src/sys/socket/sockopt.rs:784]: are you sure ? I see nginx trying to get the filter [here](https://github.com/nginx/nginx/blob/1c6183725247024f1bca73ac9a833098af7558af/src/core/ngx_connection.c#L346). Also freebsd regression tests, the expected failure is not about getsockopt [here](https://github.com/freebsd/freebsd-src/blob/676f02a4823f8a4a6e10a1cba73c6324c19b651b/tools/regression/sockets/accf_data_attach/accf_data_attach.c#L97).

Comment (asomers) [reply src/sys/socket/sockopt.rs:784]: No, looking at the kernel code I see that you're right.  I was just confused by the wording in the man page.

**Review by asomers — APPROVED** [review]
bors r+

**Pull request merged**